Source/Validation.swift

```diff
@@ -34,9 +39,117 @@ extension Request {
         case failure(Error)
     }
 
-    /// A closure used to validate a request that takes a URL request and URL response, and returns whether the
+    fileprivate struct MIMEType {
+        let type: String
+        let subtype: String
+
+        init?(_ string: String) {
+            let components: [String] = {
+                let stripped = string.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines)
```

Comment:
Slightly off topic, I believe you can remove the `CharacterSet` from `CharacterSet.whitespaceAndNewlines`. For future reference do you prefer the fully qualified types in these instances or are the static values good enough?